onnxruntime/core/providers/cuda/cuda_execution_provider.cc

```diff
@@ -359,6 +373,33 @@ Status CUDAExecutionProvider::SetComputeStream(void* stream) {
   return Status::OK();
 }
 
+#if defined(CUDA_VERSION) && CUDA_VERSION >= 10000
+void CUDAExecutionProvider::CaptureBegin()  {
+  ORT_THROW_IF_ERROR(Sync());
+  graph_.Reset();
+  graph_.CaptureBegin();
+}
+
+void CUDAExecutionProvider::CaptureEnd() {
+  graph_.CaptureEnd();
+  ORT_THROW_IF_ERROR(Sync());
```

Comment:
Do we really need this Sync() call ? We invoke this in OnRunEnd() right before the stream sync and that should mean we don't need this device sync here too ?

Reply:
This is one-time call when capturing CUDA graph, so the cost seems to be fine? It is more for safety, but I agree it seems to be not necessary.

Reply:
Agreed - it is a one-time cost and shouldn't really be a performance issue as such - but it is confusing while following the code to see syncs in multiple places when it would suffice to have one. Please consider this comment a nit.

Reply:
Superfluous syncs have been removed.